app/Utils/Traits/GeneratesCounter.php

```diff
@@ -420,8 +420,22 @@ private function checkEntityNumber($class, $entity, $counter, $padding, $pattern
         $check = false;
         $check_counter = 1;
 
+        $number_placeholder = '{number}';
+        $latest = $class::where('company_id', $entity->company_id)->where('number', '<>', '', 'and')->orderBy('created_at', 'desc')->first();
+        $generic_number = $this->applyNumberPattern($entity, $number_placeholder, $pattern);
+        $regex = sprintf('/^%s$/', preg_quote($generic_number));
+        $regex = str_replace(preg_quote($number_placeholder), '([0-9]+)', $regex);
+
+        if ($latest && preg_match($regex, $latest->number, $matches)) {
+            $latest_counter = $matches[1] ?? null;
+
+            if (is_string($latest_counter) && ctype_digit($latest_counter) && intval($latest_counter) > $counter) {
+                $counter = intval($latest_counter) + 1;
+            }
+        }
+
```

Comment:
I'm not sure we can use this. This overrides the $counter that is passed into the method. The latest counter may not always be the last record as users can delay the counter generation if this setting is selected.

Reply:
It only overwrites, if counter < latest counter. But I think this is also not a problem, because if you get a smaller counter with same rest pattern to latest counter, you have a generic problem with the counter number in settings (without using latest, we could result in conflicts (may run loop 100 times because of already used numbers & then returning rand number).

And if you don't get any latest counter, function still works as before.